pkg/utils/get_key.go

```diff
@@ -71,20 +85,39 @@ func NewAPIKey(loggerIn *zap.Logger) (*APIKeyImpl, error) {
 	return apiKeyImp, err
 }
 
-// GetIAMKeys ...
-func (d *APIKeyImpl) UpdateIAMKeys(config *config.Config) (error) {
-  if true { // IKS env
-    if config.Bluemix.Encryption {
-		    config.Bluemix.IamAPIKey = ""
-	  }
-	  if config.VPC.Encryption {
-		    if config.VPC.APIKey != "" {
-			       config.VPC.APIKey = ""
-		    }
-		    if config.VPC.G2APIKey != "" {
-			       config.VPC.G2APIKey = ""
-		    }
-	  }
-  }
-  return nil
+//UpdateIAMKeys decrypts the API keys and updates.
+func (d *APIKeyImpl) UpdateIAMKeys(config *config.Config) error {
+	//Setup grpc connection
+	var GRPCBackend grpcClient.GrpcSessionFactory
+	grpcSess := GRPCBackend.NewGrpcSession()
+	cc := &grpcClient.GrpcSes{}
+	conn, err := grpcSess.GrpcDial(cc, *endpoint, grpc.WithInsecure(), grpc.WithBlock(), grpc.WithDialer(UnixConnect))
+	if err != nil {
+		fmt.Errorf("Failed to establish grpc-client connection: %v", err)
+		return nil
+	}
+
+	//APIKeyProvider Client
+	c := pb.NewAPIKeyProviderClient(conn)
+	_, requestID := GetContextLogger(context.Background(), false)
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+	defer cancel()
```

Comment:
close the client session as well

Reply:
Done in commit "Call for closing grpc client"